modules/apps/portal-url-builder/portal-url-builder-api/src/main/java/com/liferay/portal/url/builder/ESModuleAbsolutePortalURLBuilder.java

```diff
@@ -0,0 +1,24 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.portal.url.builder;
+
+/**
+ * Builds an ES module URL.
+ *
+ * @author Iván Zaera Avellón
+ */
+public interface ESModuleAbsolutePortalURLBuilder
```

Comment:
noob java question here, but why would you create a new interface that contains nothing?

Reply:
For two reasons:

1. A marker interface: simply a marker to say that one object has certain semantics so that you can search for it with your IDE, or pass it as a parameter specifying that you want it to be of some kind.
2. So that in the future we can add new methods to objects with those semantics.